File: comguanmuonehours/src/main/java/com/guanmu/onehours/SharePreferenceUtils.java
package com.guanmu.onehours;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.Date;

/**
 * 用于保存一些基础的用户设置
 *
 * @author zengzyong 2015.7.24
 */
public class SharePreferenceUtils {

    private static final String SHARENAME = "info";
    private static final String PREF_TARGET_COUNT = "target_cont";
    private static final String FIRST_INTRO = "first_intro";
    private static final String FIRST_OPEN = "first_open";
    private static final String FIRST_EDIT = "first_edit";
    private static final String FIRST_RECITE = "first_recite";
    private static final String FIRST_KNOW = "first_know";
    private static final String LAST_UPDATE = "last_update";


    private static SharePreferenceUtils Instance = null;

    private static Context mContext;
    private SharedPreferences preferences;

    private SharePreferenceUtils() {
        preferences = mContext.getSharedPreferences(SHARENAME,
                Context.MODE_PRIVATE);
    }

    public static void init(Context context) {
        mContext = context;
    }

    public static SharePreferenceUtils getInstance() {
        if (null == Instance) {
            synchronized (SharePreferenceUtils.class) {
                if (null == Instance) {
                    Instance = new SharePreferenceUtils();
                }
            }
        }
        return Instance;
    }

    /**
     * 保存每次最大目标数
     *
     * @param count 每次最大目标数
     * @return 是否成功保存
     */
    public boolean setTargetCount(int count) {
        return preferences.edit().putInt(PREF_TARGET_COUNT, count).commit();
    }

    /**
     * 获取每次最大目标数
     *
     * @return 目标
     */
    public int getTargetCount() {
        return preferences.getInt(PREF_TARGET_COUNT, 20);
    }


    public boolean setLastUpdate() {
        return preferences.edit().putLong(LAST_UPDATE, new Date().getTime()).commit();
    }

    public Date getLastUpdate() {
        return new Date(preferences.getLong(LAST_UPDATE, 1));
    }


    public boolean getFirstIntro() {
        if (preferences.getBoolean(FIRST_INTRO, true)) {
            preferences.edit().putBoolean(FIRST_INTRO, false).apply();
            return true;
        } else {
            return false;
        }
    }


    public boolean getFirstOpen() {
        if (preferences.getBoolean(FIRST_OPEN, true)) {
            preferences.edit().putBoolean(FIRST_OPEN, false).apply();
            return true;
        } else {
            return false;
        }
    }

    public boolean getFirstEdit() {
        if (preferences.getBoolean(FIRST_EDIT, true)) {
            preferences.edit().putBoolean(FIRST_EDIT, false).apply();
            return true;
        } else {
            return false;
        }
    }

    public boolean getFirstRecite() {
        if (preferences.getBoolean(FIRST_RECITE, true)) {
            preferences.edit().putBoolean(FIRST_RECITE, false).apply();
            return true;
        } else {
            return false;
        }
    }

    public boolean getFirstKnow() {
        if (preferences.getBoolean(FIRST_KNOW, true)) {
            preferences.edit().putBoolean(FIRST_KNOW, false).apply();
            return true;
        } else {
            return false;
        }
    }

}

File: comguanmuonehours/src/main/java/com/guanmu/onehours/SharePreferenceUtils.kt
package com.guanmu.onehours

import android.content.Context
import android.content.SharedPreferences
import java.util.*

/**
 * 用于保存一些基础的用户设置
 *
 * @author zengzyong 2015.7.24
 */
class SharePreferenceUtils(mContext: Context) {
    private val preferences: SharedPreferences =
            mContext.getSharedPreferences(SHARENAME, Context.MODE_PRIVATE)

    /**
     * 获取每次最大目标数
     *
     * @return 目标
     */
    var targetCount: Int
        get() = preferences.getInt(PREF_TARGET_COUNT, 20)
        set(v) = preferences.edit().putInt(PREF_TARGET_COUNT, v).apply()

    val lastUpdate: Date
        get() = Date(preferences.getLong(LAST_UPDATE, 1))


    val firstIntro: Boolean
        get() {
            return if (preferences.getBoolean(FIRST_INTRO, true)) {
                preferences.edit().putBoolean(FIRST_INTRO, false).apply()
                true
            } else {
                false
            }
        }


    val firstOpen: Boolean
        get() {
            return if (preferences.getBoolean(FIRST_OPEN, true)) {
                preferences.edit().putBoolean(FIRST_OPEN, false).apply()
                true
            } else {
                false
            }
        }

    val firstEdit: Boolean
        get() {
            return if (preferences.getBoolean(FIRST_EDIT, true)) {
                preferences.edit().putBoolean(FIRST_EDIT, false).apply()
                true
            } else {
                false
            }
        }

    val firstRecite: Boolean
        get() {
            if (preferences.getBoolean(FIRST_RECITE, true)) {
                preferences.edit().putBoolean(FIRST_RECITE, false).apply()
                return true
            } else {
                return false
            }
        }

    val firstKnow: Boolean
        get() {
            if (preferences.getBoolean(FIRST_KNOW, true)) {
                preferences.edit().putBoolean(FIRST_KNOW, false).apply()
                return true
            } else {
                return false
            }
        }

    /**
     * 保存每次最大目标数
     *
     * @param count 每次最大目标数
     * @return 是否成功保存
     */
    fun setTargetCount(count: Int): Boolean {
        return preferences.edit().putInt(PREF_TARGET_COUNT, count).commit()
    }


    fun setLastUpdate(): Boolean {
        return preferences.edit().putLong(LAST_UPDATE, Date().time).commit()
    }

    companion object {

        private val SHARENAME = "info"
        private val PREF_TARGET_COUNT = "target_cont"
        private val FIRST_INTRO = "first_intro"
        private val FIRST_OPEN = "first_open"
        private val FIRST_EDIT = "first_edit"
        private val FIRST_RECITE = "first_recite"
        private val FIRST_KNOW = "first_know"
        private val LAST_UPDATE = "last_update"


        private lateinit var Instance: SharePreferenceUtils

        fun init(context: Context) {
            Instance = SharePreferenceUtils(context)
        }

        val instance: SharePreferenceUtils
            get() {
                return Instance
            }
    }
}
